core/meta/src/TCling.cxx

```diff
@@ -5283,8 +5283,8 @@ void TCling::UpdateClassInfoWithDecl(const void* vTD)
       }
       clang::QualType type( td->getTypeForDecl(), 0 );
 
-      auto declName=ND->getNameAsString().c_str();
-      if (!TClass::HasNoInfoOrEmuOrFwdDeclaredDecl(declName)){
+      auto declName = ND->getNameAsString();
```

Comment:
I am fixing the problem, not re-writing the code. Similar fix was done by Axel in master branch. You have to save temporary string on stack before you take (leak) it's internal pointer to actual string data. Otherwise memory is released after statement ends.